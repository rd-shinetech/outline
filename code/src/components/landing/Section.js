import React from 'react';
import reactjs from '../../img/reactjs.jpg';
import createReactApp from '../../img/createReactApp.jpg';
import reactrouter4 from '../../img/reactrouter4.jpg';
import bootstrap4 from '../../img/bootstrap4.jpg';



const Section = () => {
    return (

        <section className="row">
            <div className="col-md-6 col-lg-3">
                <div className="card">
                    <a href="https://facebook.github.io/react" target="_blank">
                        <img src={reactjs} className="card-img-top img-fluid" alt="ReactJS"/>
                    </a>
                    <div className="card-block">
                        <h3 className="card-title">ReactJS</h3>
                        <p>ReactJS/Redux using the ES2015 (ES6) syntax.</p>
                    </div>
                </div>
            </div>


            <div className="col-md-6 col-lg-3">
                <div className="card">
                    <a href="https://github.com/facebookincubator/create-react-app" target="_blank">
                        <img src={createReactApp} className="card-img-top img-fluid" alt="Create React App"/>
                    </a>
                    <div className="card-block">
                        <h3 className="card-title">create-react-app</h3>
                        <p>Official. No Setup. Built in HMR, ESLint, Jest, Coverage</p>
                    </div>
                </div>
            </div>

            <div className="col-md-6 col-lg-3">
                <div className="card">
                    <a href="https://reacttraining.com/react-router" target="_blank">
                        <img src={reactrouter4} className="card-img-top img-fluid" alt="React Router 4"/>
                    </a>
                    <div className="card-block">
                        <h3 className="card-title">React Router 4</h3>
                        <p>Declarative routing for ReactJS apps.</p>
                    </div>
                </div>
            </div>

            <div className="col-md-6 col-lg-3">
                <div className="card">
                    <a href="http://v4-alpha.getbootstrap.com" target="_blank">
                    <img src={bootstrap4} className="card-img-top img-fluid" alt="Bootstrap 4" />
                    </a>
                    <div className="card-block">
                        <h3 className="card-title">Bootstrap 4</h3>
                        <p>A framework for styling apps for all screen sizes.</p>
                    </div>
                </div>
            </div>

        </section>
    );
};



export default Section;
