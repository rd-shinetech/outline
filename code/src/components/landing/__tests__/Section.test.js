import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Section from '../Section';


describe('Section.test.js', () => {

    it('renders without crashing', () => {
        const wrapper = shallow(<Section/>);

        expect(wrapper).toHaveLength(1);

        expect(wrapper.find('section')).toHaveLength(1);
        expect(wrapper.find('.card')).toHaveLength(4);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();           
    });
});
