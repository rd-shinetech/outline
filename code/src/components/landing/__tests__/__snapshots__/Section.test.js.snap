exports[`Section.test.js renders without crashing 1`] = `
<section
  className="row">
  <div
    className="col-md-6 col-lg-3">
    <div
      className="card">
      <a
        href="https://facebook.github.io/react"
        target="_blank">
        <img
          alt="ReactJS"
          className="card-img-top img-fluid"
          src="reactjs.jpg" />
      </a>
      <div
        className="card-block">
        <h3
          className="card-title">
          ReactJS
        </h3>
        <p>
          ReactJS/Redux using the ES2015 (ES6) syntax.
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6 col-lg-3">
    <div
      className="card">
      <a
        href="https://github.com/facebookincubator/create-react-app"
        target="_blank">
        <img
          alt="Create React App"
          className="card-img-top img-fluid"
          src="createReactApp.jpg" />
      </a>
      <div
        className="card-block">
        <h3
          className="card-title">
          create-react-app
        </h3>
        <p>
          Official. No Setup. Built in HMR, ESLint, Jest, Coverage
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6 col-lg-3">
    <div
      className="card">
      <a
        href="https://reacttraining.com/react-router"
        target="_blank">
        <img
          alt="React Router 4"
          className="card-img-top img-fluid"
          src="reactrouter4.jpg" />
      </a>
      <div
        className="card-block">
        <h3
          className="card-title">
          React Router 4
        </h3>
        <p>
          Declarative routing for ReactJS apps.
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6 col-lg-3">
    <div
      className="card">
      <a
        href="http://v4-alpha.getbootstrap.com"
        target="_blank">
        <img
          alt="Bootstrap 4"
          className="card-img-top img-fluid"
          src="bootstrap4.jpg" />
      </a>
      <div
        className="card-block">
        <h3
          className="card-title">
          Bootstrap 4
        </h3>
        <p>
          A framework for styling apps for all screen sizes.
        </p>
      </div>
    </div>
  </div>
</section>
`;
